site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,462 @@
+            cy.login('instructor');
+            calculateCache();
+            cy.visit(['sample', 'gradeable', 'late_allowed_homework']);
+        });
+
+        it('Adds a new submission', () => {
+            // Make student submission
+            cy.get('#radio-student').check();
+            cy.get('#user_id').type('student');
+
+            // attatch file
+            cy.get('#input-file1').attachFile('submission.txt');
+            cy.get('#submit').click();
+
+            // Confirm dialog box
+            cy.get('#previous-submission-form')
+                .find('input')
+                .contains('Submit')
+                .click();
+
+            // Check cache
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.get('[data-user-content="student"][data-before-content="Late Allowed Homework"] ~')
+                .then((cell) => expect(cell.text().trim()).to.equal(''));
+        });
+
+        it('Cancels submission', () => {
+            // Click do not grade
+            cy.get('#do_not_grade').click();
+            // Note: page refresh triggers a late day recalulation for banner text
+
+            // Check cache
+            cy.visit(['sample', 'late_table']);
+
+            cy.get('td[data-before-content="Event/Assignment"]')
+                .contains('Late Allowed Homework')
+                .siblings('td[data-before-content="Status"]')
+                .contains('Cancelled Submission')
+                .should('exist');
+        });
+
+        it('Add gradeable version back', () => {
+            // Select gradeable
+            cy.get('#submission-version-select').select('1');
+
+            // Change the version to grade
+            cy.get('#version_change').click();
+            // Note: page refresh triggers a late day recalulation for banner text
+
+            // Check cache
+            cy.visit(['sample', 'late_table']);
+
+            cy.get('td[data-before-content="Event/Assignment"]')
+                .contains('Late Allowed Homework')
+                .siblings('td[data-before-content="Status"]')
+                .should('not.contain', 'Cancelled Submission')
+                .should('exist');
+        });
+    });
+
+    describe('Test gradable creation/deletion', () => {
+
+        beforeEach(() => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+            calculateCache();
+        });
+
+        it('Creates a gradeable', () => {
+            cy.visit(['sample', 'gradeable']);
+
+            // Enter gradeable info
+            cy.get('#g_title').type('Delete Me');
+            cy.get('#g_id').type('deleteme');
+            cy.get('#radio_ef_student_upload').check();
+            // Create Gradeable
+            cy.get('#create-gradeable-btn').click();
+
+            // Check that cache is deleted
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Delete Me"] ~')
```

Comment:
Can you find a better selector for this, and all other similar selectors in this PR?